Research/ResearchUI/iOS/RSDStepViewController.swift

```diff
@@ -1038,53 +1042,52 @@ open class RSDStepViewController : UIViewController, RSDStepController, RSDCance
-    /// timer has fired. This is **only** called if the app is running in the background.
-    /// Otherwise, the app will automatically call `goForward`.
-    open func playCompletedAlert() {
-        vibrateDevice()
-        playSound(.alarm)
-    }
-    
-    private var _activeObserver: Any?
-    private var _hasCalledGoForward: Bool = false
-    
     private func stepCompleted() {
         guard completedUptime == nil else { return }
         completedUptime = RSDClock.uptime()
-        if UIApplication.shared.applicationState == .active {
-            _goForwardOnActive()
-        } else {
-            _playAlarm()
-            _activeObserver = NotificationCenter.default.addObserver(forName: UIApplication.didBecomeActiveNotification, object: nil, queue: OperationQueue.main, using: { [weak self] (_) in
-                self?._goForwardOnActive()
-            })
-        }
+        goForward()
     }
     
-    private func _goForwardOnActive() {
-        guard !_hasCalledGoForward else { return }
-        _hasCalledGoForward = true
+    // MARK: Alert user to open the app
         
-        if let observer = _activeObserver {
-            NotificationCenter.default.removeObserver(observer)
-        }
-        goForward()
-    }
+    private var _startedAlarm: Bool = false
     
+    private func startBackgroundAlert() {
+        guard !_startedAlarm, UIApplication.shared.applicationState != .active else { return }
+        _startedAlarm = true
+        // Play the alarm after a delay.
+        let delay = DispatchTime.now() + .milliseconds(500)
+        DispatchQueue.main.asyncAfter(deadline: delay, execute: { [weak self] in
+            self?._playAlarm()
+        })
+    }
+
     private func _playAlarm() {
-        guard !_hasCalledGoForward, UIApplication.shared.applicationState != .active else { return }
-        
-        // play sound and vibrate to let the user know that the step is over
-        playCompletedAlert()
-        
-        // Fire again after a delay
+        guard UIApplication.shared.applicationState != .active else { return }
+
+        // play sound and vibrate to let the user know that the step is over.
+        playAlert()
+
+        // Fire again after a delay.
         let delay = DispatchTime.now() + .seconds(2)
         DispatchQueue.main.asyncAfter(deadline: delay, execute: { [weak self] in
             self?._playAlarm()
         })
     }
+    
+    /// If the app is running in the background, and running in the background is not allowed for this step,
+    /// then the app will start calling `playAlert()` using dispatch_async with a delay. By default, that
+    /// method will play an alarm sound and vibrate the device to alert the user to bring the app to the
+    /// foreground in order to continue the task. Once the app is active, then it will start the step.
+    ///
+    /// An alert to play if the step should transition automatically and the user has put the
```

Comment:
Is this paragraph now redundant? Either way, the first sentence here seems oddly worded or out-of-place (this isn't an alert, it's a function to play an alert).